src/graph.rs

```diff
@@ -350,12 +331,8 @@ fn hash_files(
         let mtime = file_state
             .get(id)
             .unwrap_or_else(|| panic!("no state for {:?}", graph.file(id).name));
-        let mtime_int = match mtime {
-            MTime::Missing => panic!("missing file {:?}", graph.file(id).name),
-            MTime::Stamp(t) => t,
-        };
```

Comment:
I think we want to preserve this panic, since it's an invariant.

I think you can just hash the SystemTime itself rather than the MTime wrapper, and remove Hash from MTime.

Reply:
Ok done.